src/bin/uniq.rs

```diff
@@ -0,0 +1,163 @@
+#![deny(warnings)]
+
+extern crate coreutils;
+extern crate extra;
+
+use std::env;
+use std::mem;
+use std::fs::File;
+use std::process::exit;
+use std::io::{stdout, stderr, stdin, Error, Write, BufRead, BufReader};
+use coreutils::ArgParser;
+use extra::option::OptionalExt;
+
+const MAN_PAGE: &'static str = r#"
+NAME
+    uniq - report or omit repeated lines
+
+SYNOPSIS
+    uniq [ -h | --help | -d | -u | -i ] [FILE]...
+
+DESCRIPTION
+    Filter adjacent matching lines of FILE(s) to standard output.
+
+    With no FILE(s), read standard input.
+
+OPTIONS
+    -h
+    --help
+        display this help and exit
+    -c
+    --count
+        precede each output line with a count of the number of times the line occurred in the input
+    -i
+    --ignore-case
+        compare lines case-insensitively
+    -d
+    --repeated-lines
+        only print duplicate lines, one for each group
+    -u
+    --unique-lines
+        only print unique lines
+"#; /* @MANEND */
+
+fn lines_from_stdin() -> Result<Vec<String>, Error> {
+    let stdin = stdin();
+    let mut lines = Vec::new();
+
+    let f = BufReader::new(stdin.lock());
+    for line in f.lines() {
```

Comment:
Regarding the case-insensitive comparison, are these optimisations still valid if the input is not pure ascii? It would only compare ascii characters in a case-insensitive manner, but other characters would not be compared properly, like é and É. If the goal is to match, say, GNU uniq's behavior, then this is ok. If (for some reasons) we want to properly support unicode, it's not valid.